rustls/src/crypto/ring/sign.rs

```diff
@@ -53,20 +55,16 @@ pub fn any_ecdsa_type(der: &PrivateKeyDer<'_>) -> Result<Arc<dyn SigningKey>, In
         return Ok(Arc::new(ecdsa_p384));
     }
 
-    Err(InvalidKeyError(()))
+    Err(Error::General(
+        "failed to parse ECDSA private key as PKCS#8 or SEC1".into(),
+    ))
 }
 
 /// Parse `der` as any EdDSA key type, returning the first which works.
-pub fn any_eddsa_type(
-    der: &PrivatePkcs8KeyDer<'_>,
-) -> Result<Arc<dyn SigningKey>, InvalidKeyError> {
-    if let Ok(ed25519) = Ed25519SigningKey::new(der, SignatureScheme::ED25519) {
-        return Ok(Arc::new(ed25519));
-    }
-
+pub fn any_eddsa_type(der: &PrivatePkcs8KeyDer<'_>) -> Result<Arc<dyn SigningKey>, Error> {
     // TODO: Add support for Ed448
-
-    Err(InvalidKeyError(()))
+    let ed25519 = Ed25519SigningKey::new(der, SignatureScheme::ED25519)?;
```

Comment:
Nit: I think inlining the one-use trivial `ed25519` binding here would be more idiomatic.